### DISCUSSION TITLE
Matlab's ncread is so convenient

### DISCUSSION THREAD

**Discussion by hugary1995** [idaholab/moose#17852] [Q&A General] [answered]
I just accidentally deleted an input file, and Matlab's ncread function saved me once again. I hope everyone is aware of this feature.

If you still have access to your output exodus file. In Matlab, you can do
`ncread('out.e', 'info_records')'`
to get back the input file! Notice the transpose operation in the end.

I believe you can use `scipy.io.netcdf` to do the same thing but your mileage may vary.

**Comment by Traiwit** [comment, marked as answer]
Awesome, thanks @hugary1995

**Comment by GiudGiud** [comment]
In MOOSE we have --show-input that will print the input at the beginning of your output log. 
Not as good as getting it from exodus for sure.
